tests/test_iqgen_gen_main.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (C) 2016 Swift Navigation Inc.
+#
+# Contact: Valeri Atamaniouk <valeri@swiftnav.com>
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+'''
+Unit tests for IQgen generator main function
+'''
+
+from peregrine.iqgen.generate import generateSamples
+from peregrine.iqgen.bits.satellite_gps import GPSSatellite
+from peregrine.iqgen.bits.satellite_glo import GLOSatellite
+from peregrine.iqgen.if_iface import HighRateConfig
+from peregrine.iqgen.bits.encoder_other import GPSGLONASSBitEncoder
+
+
+def test_generateSamples0():
+  '''
+  Sample generation test:
+  - GPS L1/L2 + GLONASS L1/L2
+  - Noise sigma defined
+  - Band pass filter type
+  - No group delays
+  - No process spawning
+  '''
```

Comment:
Add a comment saying that the test is supposed to fail on an exception during the samples generation, if something is wrong.